osbuild-composer.spec

```diff
@@ -155,8 +155,21 @@ install -m 0755 -vp _bin/osbuild-composer                       %{buildroot}%{_l
 install -m 0755 -vp _bin/osbuild-worker                         %{buildroot}%{_libexecdir}/osbuild-composer/
 install -m 0755 -vp dnf-json                                    %{buildroot}%{_libexecdir}/osbuild-composer/
 
+# Only include repositories for the distribution and release
 install -m 0755 -vd                                             %{buildroot}%{_datadir}/osbuild-composer/repositories
-install -m 0644 -vp repositories/*                              %{buildroot}%{_datadir}/osbuild-composer/repositories/
+# CentOS also defines rhel so we check for centos first
+%if 0%{?centos}
+install -m 0644 -vp repositories/centos-%{centos}*              %{buildroot}%{_datadir}/osbuild-composer/repositories/
+install -m 0644 -vp repositories/centos-stream-%{centos}*       %{buildroot}%{_datadir}/osbuild-composer/repositories/
+%else
+%if 0%{?rhel}
+install -m 0644 -vp repositories/rhel-%{rhel}*                  %{buildroot}%{_datadir}/osbuild-composer/repositories/
```

Comment:
We probably do want to allow building RHEL8 on a RHEL9 host. I.e., backwards compatibility should be fine, forwards compatibility would be harder (i.e., doesn't work, so better not expose it).